File: sdk/elixir/lib/rocksky/googledrive.ex
defmodule Rocksky.GoogleDrive do
  @moduledoc "`app.rocksky.googledrive.*` endpoints. Require an authenticated client."

  alias Rocksky.HTTP

  @doc "List files at a Drive path. Params: `:at`."
  def get_files(client, params \\ []),
    do: HTTP.query(client, "app.rocksky.googledrive.getFiles", params)

  @doc "Get a single file metadata. Params: `:fileId`."
  def get_file(client, params \\ []),
    do: HTTP.query(client, "app.rocksky.googledrive.getFile", params)

  @doc "Download a file by id. Params: `:fileId`."
  def download_file(client, params \\ []),
    do: HTTP.query(client, "app.rocksky.googledrive.downloadFile", params)
end

File: sdk/clojure/src/rocksky/googledrive.clj
(ns rocksky.googledrive
  "Endpoints under app.rocksky.googledrive.*"
  (:require [rocksky.client :as c]))

(defn get-files
  "List files in Google Drive. Optional: `:at` (folder path)."
  ([client] (get-files client nil))
  ([client {:keys [at]}]
   (c/query client :app.rocksky.googledrive.getFiles {:at at})))

(defn get-file
  "Get a single file by `:file-id`. Required: `:file-id`."
  [client {:keys [file-id]}]
  (c/query client :app.rocksky.googledrive.getFile {:fileId file-id}))

(defn download-file
  "Download a file by `:file-id`. Required: `:file-id`."
  [client {:keys [file-id]}]
  (c/query client :app.rocksky.googledrive.downloadFile {:fileId file-id}))
